### DISCUSSION TITLE
Using directives in need_count

### DISCUSSION THREAD

**Discussion by bokzzzz** [useblocks/sphinx-needs#1180] [Q&A]
Hello, I was using spinix-needs 2.0.0 and I had in doc one need count: ``:need_count:`type=='spec' and docname.startswith('software_') and satisfies`` and it works fine, but when I upgraded to 2.1.0 version I got error like this: 
```
Filter "type=='spec' and docname.startswith('software_') and satisfies" not valid. 
Error: Filter did not evaluate to a boolean, instead <class 'list'>: []. [needs.filter]. 
```
Can someone help me what I need to change in my need_count to work with 2.1.0 version?

**Comment by danwos** [comment]
The used filter string does not return a boolean, which is needed by Sphinx-Needs to check if a need shall be part of the result or not.

You are checking the content of a list by simply `and satisfies`. This returns an empty list and not a boolean value. So in the end your filter strings result may look like `True and []`, which is then a list.

Solution is simple, just check for the length: `and len(satisfies)>0`.
